src/draw/espressif/ppa/lv_draw_ppa.c

```diff
@@ -0,0 +1,239 @@
+
+    if(!ppa_dest_cf_supported(base->layer->color_format)) return 0;
+
+    switch(t->type) {
+        case LV_DRAW_TASK_TYPE_FILL: {
+                const lv_draw_fill_dsc_t * dsc = (lv_draw_fill_dsc_t *)t->draw_dsc;
+                if((dsc->radius != 0 || dsc->grad.dir != LV_GRAD_DIR_NONE)) return 0;
+
+                if(t->preference_score > DRAW_UNIT_PPA_PREF_SCORE) {
+                    t->preference_score = DRAW_UNIT_PPA_PREF_SCORE;
+                    t->preferred_draw_unit_id = DRAW_UNIT_ID_PPA;
+                }
+                return 1;
+            }
+
+#if LV_USE_PPA_IMG
+        case LV_DRAW_TASK_TYPE_IMAGE: {
+                lv_draw_image_dsc_t * dsc = t->draw_dsc;
+                if(!(dsc->header.cf < LV_COLOR_FORMAT_PROPRIETARY_START
+                     && dsc->clip_radius == 0
+                     && dsc->bitmap_mask_src == NULL
+                     && dsc->sup == NULL
+                     && dsc->tile == 0
+                     && dsc->blend_mode == LV_BLEND_MODE_NORMAL
+                     && dsc->recolor_opa <= LV_OPA_MIN
+                     && dsc->skew_y == 0
+                     && dsc->skew_x == 0
+                     && dsc->scale_x == 256
+                     && dsc->scale_y == 256
+                     && dsc->rotation == 0
+                     && lv_image_src_get_type(dsc->src) == LV_IMAGE_SRC_VARIABLE
+                     && (dsc->header.cf == LV_COLOR_FORMAT_ARGB8888
+                         || dsc->header.cf == LV_COLOR_FORMAT_XRGB8888
+                         || dsc->header.cf == LV_COLOR_FORMAT_RGB888
+                         || dsc->header.cf == LV_COLOR_FORMAT_RGB565)
+                     && (dsc->base.layer->color_format == LV_COLOR_FORMAT_ARGB8888
+                         || dsc->base.layer->color_format == LV_COLOR_FORMAT_XRGB8888
+                         || dsc->base.layer->color_format == LV_COLOR_FORMAT_RGB888
+                         || dsc->base.layer->color_format == LV_COLOR_FORMAT_RGB565))) {
+                    return 0;
+                }
+
+                if(t->preference_score > DRAW_UNIT_PPA_PREF_SCORE) {
+                    t->preference_score = DRAW_UNIT_PPA_PREF_SCORE;
+                    t->preferred_draw_unit_id = DRAW_UNIT_ID_PPA;
+                }
+                return 1;
+            }
+#endif
+        default:
+            return 0;
+    }
+}
+
+static int32_t ppa_dispatch(lv_draw_unit_t * draw_unit, lv_layer_t * layer)
+{
+    lv_draw_ppa_unit_t * u = (lv_draw_ppa_unit_t *)draw_unit;
+    if(u->task_act) return 0;
+
+    lv_draw_task_t * t = lv_draw_get_available_task(layer, NULL, DRAW_UNIT_ID_PPA);
+    if(!t || t->preferred_draw_unit_id != DRAW_UNIT_ID_PPA) return LV_DRAW_UNIT_IDLE;
+    if(!lv_draw_layer_alloc_buf(layer)) return LV_DRAW_UNIT_IDLE;
+
+    t->state = LV_DRAW_TASK_STATE_IN_PROGRESS;
+    u->task_act = t;
+    u->task_act->draw_unit = draw_unit;
+
+    ppa_execute_drawing(u);
+
+#ifndef LV_PPA_NONBLOCKING_OPS
```

Comment:
```suggestion
#if !LV_PPA_NONBLOCKING_OPS
```

Also, i don't see this config anywhere. Is this feature not complete yet and thus the config is not yet available? If that's the case we need to define this in the `lv_draw_ppa_private.h` as 0 for now

Reply:
Yes this config is not something ready yet until we get proper understanding of the PPA hardware, so I preferred to not define it to not enforce the usage on the LVGL default configuration even in cases where the PPA draw unit is enabled.

EDIT: Actually it is a good idea to hard define it to 0 on the private file, it will avoid the user to set this option to 1 since we can emit a compiler error

Reply:
I figured and I agree to keep it private for now.
Something needs to change tho. In the rest of the code you're using `#if`. I see two options:
1. Either define it to `0` in the private header. The config stays private but the symbol is actually defined
2. Replace all others `#if` with `#ifdef` but with the maintenance burden of having to modify everything back to `#if` when this config eventually gets added

Maybe i'm missing something tho, what do you think ?